base/linalg/diagonal.jl

```diff
@@ -101,9 +101,9 @@ end
 parent(D::Diagonal) = D.diag
 
 ishermitian{T<:Real}(D::Diagonal{T}) = true
-ishermitian(D::Diagonal) = all(D.diag .== real(D.diag))
+ishermitian(D::Diagonal) = isreal(D.diag)
```

Comment:
is that slated to be deprecated in favor of explicit vectorization / reduction? I would say the same for iszero, might make sense to only define the scalar method